code/game/objects/structures/ladders.dm

```diff
@@ -50,31 +50,27 @@
 	else	//wtf make your ladders properly assholes
 		icon_state = "ladder00"
 
-/obj/structure/ladder/proc/go_up(mob/user,is_ghost)
+/obj/structure/ladder/proc/travel(going_up, mob/user, is_ghost, obj/structure/ladder/ladder)
 	if(!is_ghost)
-		show_fluff_message(1,user)
-		up.add_fingerprint(user)
-	user.loc = get_turf(up)
-
-/obj/structure/ladder/proc/go_down(mob/user,is_ghost)
-	if(!is_ghost)
-		show_fluff_message(0,user)
-		down.add_fingerprint(user)
-	user.loc = get_turf(down)
+		show_fluff_message(going_up,user)
+		ladder.add_fingerprint(user)
+	user.forceMove(get_turf(ladder))
+	if(user.pulling)

```

Comment:
you need to do this before moving the user, and you should also reestablish the pull after both move